_source/_posts/2023-04-19-api-integrations.md

```diff
@@ -0,0 +1,112 @@
+
+You'll create the machine-to-machine process, test it locally, and then submit the integration to the OIN for public consumption. Okta customer administrators will see all available API Service Integrations in the Okta Admin Console. They can find the integration they want to add, review the requested scopes, and authorize it. They receive a Client ID and Client Secret that they provide to you, the integration provider, along with the Okta Domain so you can make the calls to retrieve the access token.
+
+**Considerations with using API Integration Services**
+
+You'll use standard OAuth flows, so you'll have a secure authorization mechanism, but it is a 2-step process to get the access token. You'll also need to define the scopes you need beforehand, which will help customers feel more comfortable using your integration.
+
+**When to use API Integration Services**
+
+Use API Integration Services when you create machine-to-machine integration for general use within the OIN. You'll have security, discoverability, and clarity for the users when you use this approach.
+
+## Decide which API authorization mechanism to use
+
+When calling Okta APIs, we recommend using API Integration Services first, then OAuth, and finally API Tokens as shown in the following flowchart:
+
+{% img blog/api-integrations/flowchart.jpg alt:"If you are building a public integration, use API Services Integration, otherwise use OAuth. Lastly use an API Token if there are no other options." width:"800" %}{: .center-image }
+
+## Learn more about connecting to Okta APIs using OAuth Client Credentials
+
+API tokens are the simplest, but this simplicity makes them too limited for production applications' security and reliability needs. OAuth client credentials require more understanding to set up but scale indefinitely for internal use. And for code intended for use outside your organization, API Service Integrations built on top of OAuth maximize flexibility and control.
+
+If you found this post interesting, you might find these resources useful:
+
+* [Secure Your .NET 6 Web API](/blog/2022/04/20/dotnet-6-web-api)
+* [How to Use Client Credentials Flow with Spring Security](/blog/2021/05/05/client-credentials-spring-security)
+* [Implement authorization by grant type](https://developer.okta.com/docs/guides/implement-grant-type/clientcreds/main/)
+* [Build an API service integration](https://developer.okta.com/docs/guides/build-api-integration/main/)
```

Comment:
This might be another good link to add:
[API service integrations in the OIN](https://developer.okta.com/docs/guides/oin-api-service-overview/)